src/mgr/ActivePyModules.cc

```diff
@@ -420,15 +420,13 @@ void ActivePyModules::notify_all(const LogEntry &log_entry)
 bool ActivePyModules::get_config(const std::string &module_name,
     const std::string &key, std::string *val) const
 {
-  PyThreadState *tstate = PyEval_SaveThread();
-  Mutex::Locker l(lock);
-  PyEval_RestoreThread(tstate);
-
   const std::string global_key = PyModuleRegistry::config_prefix
     + module_name + "/" + key;
 
   dout(4) << __func__ << "key: " << global_key << dendl;
 
+  Mutex::Locker l(lock);
```

Comment:
mimic has `Mutex::Locker lock(module_config.lock);` here - is `l(lock)` the luminous version of that?

Reply:
@smithfarm In luminous PyModuleConfig (the type of module_config (post luminous only) and config_cache) is a std::map and does not have its own lock. Post luminous PyModuleConfig was converted to a class object with its own lock so we take the more global "ActivePyModules::lock" here.